### PR TITLE
Exceptions for large numbers instead of TypeErrors

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
Did not deal with newFromFloat yet.
For floats there is PHP_FLOAT_DIG, added in PHP 7.2

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
I had a few minutes so I poked at this after seeing some ticket about it earlier today. Don't have the link to the ticket here though.

**Comment by JeroenDeDauw** [comment]
Tests passed. Issue: master build fails due to PHPCS too old

File: src/Euro.php [hunk 1]
```diff
@@ -105,6 +113,10 @@ public static function newFromFloat( float $euroAmount ): self {
 	 * @throws InvalidArgumentException
 	 */
 	public static function newFromInt( int $euroAmount ): self {
+		if ( $euroAmount > floor( PHP_INT_MAX / 101 ) ) {
```
Comment (tzhelyazkova) [inline src/Euro.php]: why divide by 101?
The best I could find after googling is [the wikipedia article on palindromic primes](https://en.wikipedia.org/wiki/Palindromic_prime)

Comment (JeroenDeDauw) [reply src/Euro.php]: I expected the /100 to work but it does not. Hence 101. That likely does not allow representing certain big numbers that strictly speaking can be represented though that seems like something that does not actually matter here. There might also be some small range of numbers for which the type error still happens, though again these will be very large ones that don't really matter. I expect this maybe-not-100%-correct-but-definitely-more-correct-than-before change to take care of all errors that currently end up in the log.

File: src/Euro.php [hunk 2]
```diff
@@ -107,16 +108,26 @@ public static function newFromFloat( float $euroAmount ): self {
...
 
+	/**
+	 * @param int|float $euroAmount
+	 */
+	private static function assertMaximumValueNotExceeded( $euroAmount ): void {
+		// When $euroAmount == PHP_INT_MAX / self::CENTS_PER_EURO, multiplying it
+		// by self::CENTS_PER_EURO still exceeds PHP_INT_MAX, which leads type errors
+		// due to float conversion. The safest thing to do here is using a safety
+		// margin of 1 with self::CENTS_PER_EURO
+		if ( $euroAmount > floor( PHP_INT_MAX / ( self::CENTS_PER_EURO + 1 ) ) ) {
```
Comment (JeroenDeDauw) [inline src/Euro.php]: Such readability! Indeed better than 101

**Pull request merged**